code-challenges.js

```diff
@@ -69,7 +116,45 @@ var vowelTester2 = "throw"
 var websites = ["codepen.io", "codecademy.com", "coursera.org", "codepen.io", "udemy.com", "pluralsight.com", "udacity.com", "sitepoint.com"]
 // Expected output: "codepen.io" "codepen.io"
 
+const findIoSites = (array) => {
 
+  // array to hold our found io sites
+  let ioSitesArray = []
+
+  // map over the array and access each website
+  array.map(currentSite => {
+
+    let vowelIndex = 0
+    // first find the index of the '.'
+    for(let i = 0; i < currentSite.length; i++) {
+      // if you find the '.' and the next indexed value is 'i'
+      if (currentSite[i] === '.' && currentSite[i + 1] === 'i') {
+        // assume we found an .io website
+        ioSitesArray.push(currentSite)
+        break
+      } // end if
+    } // end for
+  }); // end .map
+
+  return ioSitesArray
+}
```

Comment:
This is great, but I would look into how Map already performs a loop for us, thus rendering the nested loop unnecessary.